File: CrackingTheCodeInterview/C#/Chapter1/P1 UniqueChars.cs
using System;
namespace Chapter1
{
    class UniqueChars
    {
        public static bool IsUnique(string phrase)
        {
            phrase = phrase.ToLower();
            Console.WriteLine("=======" + phrase);
            int[] abc = new int[27];
            for (int i = 0; i < phrase.Length; i++)
            {
                int value = (int)phrase[i];
                int pos = value - 96;
                // check for space
                if (value - 32 == 0)
                {
                    abc[0]++;
                    if (abc[0] > 1)
                    {
                        return false;
                    }
                    continue;
                }
                // check for letters
                if (pos > 0 && pos < 27)
                {
                    abc[pos]++;
                    if (abc[pos] > 1)
                    {
                        return false;
                    }
                }
            }
            return true;
        }

        public static void Main()
        {
            Console.WriteLine(IsUnique("Borrador"));
            Console.WriteLine(IsUnique("PIEZA"));
            Console.WriteLine(IsUnique("Es una frase"));
            Console.WriteLine(IsUnique("PUEDE serquesibrother"));
            Console.WriteLine(IsUnique("murcielago"));
        }

    }

}

File: CrackingTheCodeInterview/Java/Chapter1/P1_UniqueChars.java

class UniqueChars {
    public static boolean hasUnique(String phrase){
        phrase = phrase.toLowerCase();
        boolean[] letters = new boolean[27];
        int phrase_lenght = phrase.length();
        for (int i = 0; i< phrase_lenght ; i++ ) {
            int position = phrase.charAt(i) - 'a';
            if(letters[position]){
                return false;
            }
            letters[position] = true;
        }
        return true;
    }
    public static void main(String[] args) {
        System.out.println("Main Function");
        System.out.println(hasUnique("murcielago"));
        System.out.println(hasUnique("ana"));
        System.out.println(hasUnique(""));
        System.out.println(hasUnique("mexico"));
        System.out.println(hasUnique("internet"));
    }
}
